### PR TITLE
Csv ip

### PR DESCRIPTION
**Pull request by ghost**
I noticed you were using commander so I figured you wanted to make this a command line tool. I added a bin object to your package.json file and named the command "csv-IP".

I made a mockup .csv file and noticed you had specific edge case fields like "IPAdress". I figured out that your program plucks that property from the csvconverter stream to fetch data from the ip-api. I ran a test on a mockup with the "IPAdress"  field with some dummy data and it worked except for a couple of  problems. On line 48 in program.js:

``` javascript
function doSomething(jsonData) {          
-    cursor++;      
     if (jsonData.length === cursor) {           
       clearInterval(handle);              
       handle = 0;             
     } else {           
-      getItem(jsonData[cursor]);       
     }           
   }         
 }
```

The problem is `cursor++` is incrementing from 0 to 1 before getItem is called, this means the first iteration of getItem is going to grab data from the first index of the jsonData array, ex: `jsonData[1]`.

I changed the increment to happen during the getItem function call. This fixes that problem.

I also added the csv module to the package.json file, you required it in `lib/program.js`, i'm not sure if you're using it but it threw an error during npm install. I added the module to package.json so it's a dependency.

I also added the mkdirp module which allows you to create a directory path with parent directories ex: `data/output`. This allows you to run the command line tool anywhere in the file system (with the exception of certain permissions). If the directory path already exists, everything is ignored.

So basically now you can run the command `csv-IP -f <.csvFile in current directory>` and it will generate a data/output directory path in the current directory with the output contents written to `data/output/IpInfo.csv`

Of course for this to work, cd into the repo, `npm install` your dependencies and run `npm link` to symlink the command to your bin. You can now run `csv-IP` anywhere there is a .csv file and it will generate the output directories/file.

### CODE REVIEW

**Comment by curiouslychase** [comment]
Thanks for doing that. This was never meant to be anything more than a quick solution to a quick problem, but I appreciate your pull request. Would you mind updating the `README.md` to reflect your changes for how to run it?

**Comment by ghost** [comment]
Yeah I added a new `README.md` I figure since this use is so limited, there is probably no need to publish this to npm, however if you did, I added a new command-line option for an IP address fieldname.

Your code originally only supported "IPAddress" as the field name so I added a -k <fieldname> option, so any .csv file can be read and the ip address field name can be plucked. I put an example in the `README.md`. If this is too confusing, no worries, your code helped me implement what I needed to be done. Don't feel obligated to merge this pull request, haha.
